src/main/java/org/jenkinsci/plugin/gitea/client/impl/DefaultGiteaConnection.java

```diff
@@ -1016,7 +998,7 @@ private int delete(UriTemplate template) throws IOException, InterruptedExceptio
         }
     }
 
-    private static HttpURLConnection openConnection(UriTemplate template) throws IOException {
+    protected HttpURLConnection openConnection(UriTemplate template) throws IOException {
```

Comment:
```suggestion
    @Restricted(NoExternalUse.class)
    protected HttpURLConnection openConnection(UriTemplate template) throws IOException {
```

To hold it back from being exposed API to downstream plugins